s3transfer/utils.py

```diff
@@ -567,3 +574,57 @@ def release(self, tag, acquire_token):
                     "%s for tag: %s" % (sequence_number, tag))
         finally:
             self._condition.release()
+
+
+class ChunksizeAdjuster(object):
+    def __init__(self, max_size=MAX_SINGLE_UPLOAD_SIZE,
+                 min_size=MIN_UPLOAD_CHUNKSIZE, max_parts=MAX_PARTS):
+        self.max_size = max_size
+        self.min_size = min_size
+        self.max_parts = max_parts
+
+    def adjust_chunksize(self, current_chunksize, file_size=None):
```

Comment:
That makes sense. It would be good to document what ``None`` implies here though.